### ISSUE TITLE
Exception of type System.OutOfMemoryException

### ISSUE THREAD

**Issue by KoradaRajesh** [microsoft/BotFramework-Composer#8534]
We have built a few dialogs in Composer 1.3.1 version and hosted them in Azure and it's randomly throwing **System.OutOfMemoryException**

**Strack trace:** 
   at Microsoft.Bot.Builder.Dialogs.Declarative.Resources.ResourceExplorer.LoadTypeAsync[T](Resource resource, CancellationToken cancellationToken)     at Microsoft.Bot.Builder.Dialogs.Declarative.Resources.ResourceExplorer.LoadType[T](Resource resource)     at AIP.BotEngine.Classes.AdaptiveDialogProcessor.LoadRootDialogAsync(ResourceExplorer resourceExplorer) in D:\CCoreRepo\RajeshK\CCoreV4\AIP.BotEngine\AIP.BotEngine\Classes\AdaptiveDialogProcessor.cs:line 123     at AIP.BotEngine.Classes.AdaptiveDialogProcessor.ProcessAdaptiveDialogs(ITurnContext turnContext, ICacheManager cacheManager, UserBE userBE, ConversationState conversationState, UserState userState, ResourceExplorer resourceExplorer, CancellationToken cancellationToken) in D:\CCoreRepo\RajeshK\CCoreV4\AIP.BotEngine\AIP.BotEngine\Classes\AdaptiveDialogProcessor.cs:line 112
   
 
If this is a known issue Is this issue still exist in the latest version?

**Comment by gabog** [comment]
Hi @srinaath, do you know if this has been addressed in composer 2.1?

**Comment by tonyanziano** [comment]
Hi @KoradaRajesh ,

I searched our past issues for the term "System.OutOfMemoryException" and couldn't find any results, so I don't believe this was a known issue, but @srinaath feel free to chime in if I'm wrong.

For now I would recommend using Composer 2.x to to create a replica of the dialogs responsible for throwing the errors, and publish them to Azure and see if the problem goes away.

**Comment by srinaath** [comment]
Thanks @tonyanziano. Agreed, this is not a known issue.
@KoradaRajesh Just to clarify the bot you have runs locally fine on Composer and when deployed to Azure, running the deployed bot throws the error? Also, does the error get generated starting the bot up or when you trigger a particular dialog? Are you getting an exception when you deploy a bot built from one of the default templates?

With 1.3.1 we had used a different runtime package. In 2.0 we had made tons of improvements to the way we leverage the runtime. We have a migration step built into Composer to migrate your bot assets to the new runtime. Do go ahead and give it a try.
